File: 企查查/qcc.js
var crypto_js = require("crypto-js")

var window = global

function encode_HmacSHA512(text, key) {
    return crypto_js.HmacSHA512(text, key).toString();
}


var a_default = {
    "n": 20,
    "codes": {
        "0": "W",
        "1": "l",
        "2": "k",
        "3": "B",
        "4": "Q",
        "5": "g",
        "6": "f",
        "7": "i",
        "8": "i",
        "9": "r",
        "10": "v",
        "11": "6",
        "12": "A",
        "13": "K",
        "14": "N",
        "15": "k",
        "16": "4",
        "17": "L",
        "18": "1",
        "19": "8"
    }
}

function o_default() {
    for (var e = (arguments.length > 0 && void 0 !== arguments[0] ? arguments[0] : "/").toLowerCase(), t = e + e, n = "", i = 0; i < t.length; ++i) {
        var o = t[i].charCodeAt() % a_default.n;
        n += a_default.codes[o]
    }
    return n
}

function get_i() {
    var e = arguments.length > 1 && void 0 !== arguments[1] ? arguments[1] : {}
        , t = (arguments.length > 0 && void 0 !== arguments[0] ? arguments[0] : "/").toLowerCase()
        , n = JSON.stringify(e).toLowerCase();
    return encode_HmacSHA512(t + n, o_default(t)).toLowerCase().substr(8, 20)
}


function get_l() {
    var e = arguments.length > 1 && void 0 !== arguments[1] ? arguments[1] : {}
        , t = arguments.length > 2 && void 0 !== arguments[2] ? arguments[2] : ""
        , n = (arguments.length > 0 && void 0 !== arguments[0] ? arguments[0] : "/").toLowerCase()
        , i = JSON.stringify(e).toLowerCase();
    return encode_HmacSHA512(n + "pathString" + i + t, o_default(n))
}


function s_default() {
                var list = ["w", "i", "n", "d", "o", "w", ".", "t", "i", "d"]; //window.tid
                return eval(list.join(""))
            }


function get_pageinfo(page) {
    var data = {
        "searchKey": "小米",
        "pageIndex": page,
        "pageSize": 20
    }
    var headers = {}
    var t = "/api/search/searchmulti"
    var i = get_i(t, data)
    var l = get_l(t, data, '1177adc16d2167cb61ca042cb1bb5f13'); //字符串由s_default得来：window.tid，需要访问网页文档拿到该值
    headers['key'] = i;
    headers['value'] = l;
    return headers;
}

// console.log(get_pageinfo(1))

File: 企查查/qcc.py
import execjs
import json
import requests

with open('qcc.js','r',encoding='utf-8') as f:
    js_code = f.read()
ctx = execjs.compile(js_code)

page = 1
headers_info = ctx.call('get_pageinfo',page)
print(headers_info)

headers = {
    "accept": "application/json, text/plain, */*",
    "accept-language": "zh-CN,zh;q=0.9,en-US;q=0.8,en;q=0.7",
    "cache-control": "no-cache",
    "content-type": "application/json",
    headers_info['key']: headers_info['value'],
    "origin": "https://www.qcc.com",
    "pragma": "no-cache",
    "priority": "u=1, i",
    "referer": "https://www.qcc.com/web/search?key=%E5%B0%8F%E7%B1%B3",
    "sec-ch-ua": "\"Not=A?Brand\";v=\"99\", \"Google Chrome\";v=\"151\", \"Chromium\";v=\"151\"",
    "sec-ch-ua-mobile": "?0",
    "sec-ch-ua-platform": "\"Windows\"",
    "sec-fetch-dest": "empty",
    "sec-fetch-mode": "cors",
    "sec-fetch-site": "same-origin",
    "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/151.0.0.0 Safari/537.36",
    "x-pid": "e433881c144eb274bc10020728d4bbf2",
    "x-requested-with": "XMLHttpRequest"
}
# 登录态 Cookie 已脱敏，运行时请从浏览器登录后复制真实值。
cookies = {
    "QCCSESSID": "",
    "qcc_did": "",
    "UM_distinctid": "",
    "_c_WBKFRo": "",
    "_nb_ioWEgULi": "",
    "acw_tc": "",
    "CNZZDATA1254842228": "",
}
url = 'https://www.qcc.com/api/search/searchMulti'
data = json.dumps({"searchKey": "小米", "pageIndex": page, "pageSize": 20}).encode('utf-8')
response = requests.post(url, headers=headers, cookies=cookies, data=data)

print(response.text)
print(response)